nav2_costmap_2d/include/nav2_costmap_2d/costmap_layer.hpp

```diff
@@ -192,6 +194,23 @@ class CostmapLayer : public Layer, public Costmap2D
  */
   CombinationMethod combination_method_from_int(const int value);
 
+  /**
+   * Joins the specified topic with the parent namespace of the costmap node.
+   * If the topic has an absolute path, it is returned instead.
+   *
+   * This is necessary for user defined relative topics to work as expected since costmap layers
+   * add a an additional `costmap_name` namespace to the topic.
+   * For example:
+   *   * User chosen namespace is `tb4`.
+   *   * User chosen topic is `scan`.
+   *   * Costmap node namespace will be `/tb4/global_costmap`.
+   *   * Without this function, the topic would be `/tb4/global_costmap/scan`.
+   *   * With this function, topic will be remapped to `/tb4/scan`.
```

Comment:
/+ 

```
   * Use global costmap `/scan` if you do not wish the node namespace to apply
```

Reply:
https://github.com/ros-navigation/navigation2/pull/4715/commits/ac5d861dbf44f0c80bd624c4d55722e120c12c28 (changed `Use global costmap` to `Use global topic` since `/scan` is not really a costmap?)